r2/r2/public/static/js/reddit.js

```diff
@@ -1229,6 +1229,17 @@ $(function() {
             $("#moresearchinfo").slideUp();
             event.preventDefault();
         });
+
+        /* Add a scheme to URL input fields if not already present */
+        $("input[type=url]").blur(function() {
+            var input = $(this).val();
+            if (input && input.length > 0) {
+                var schemePresent = input.match(/^[a-z]+:\/\/.*/i);
+                if (!schemePresent) {
+                    $(this).val("http://" + input);
```

Comment:
There's a part of me that wants to default to https, even if there are a bunch of sites that don't support it. But that's probably a bad idea. :)